apps/ecs/behaviours/check-validation.js

```diff
@@ -14,17 +14,33 @@ module.exports = superclass => class extends superclass {
       }
     }
 
-    if(key === 'before-1988-worker-nationality' || key === 'worker-country') {
+    if(key === 'before-1988-worker-nationality' || key === 'worker-country' ||
+     key === 'worker-nationality') {
       if(req.form.values[key] === 'United Kingdom' || req.form.values[key] === 'Ireland') {
         return validationErrorFunc('excludeUkIr');
       }
     }
 
+    if(key === 'before-1988-worker-dob') {
+      const workerDob = req.form.values[key];
+      if (moment(workerDob).isSameOrAfter('1988-01-01')) {
+        return validationErrorFunc('afterDobYear');
+      }
+    }
+
     if(key === 'before-1988-worker-year-of-entry-to-uk') {
       const yearOfEntry = req.form.values[key];
+      const oneHundredTwentyYearsAgo = moment().subtract(120, 'years').format('YYYY');
+      if (yearOfEntry && yearOfEntry <= oneHundredTwentyYearsAgo) {
+        return validationErrorFunc('after120Years');
+      }
+      if (yearOfEntry && yearOfEntry >= '1988') {
```

Comment:
`yearOfEntry >= '1988'` is a string comparison, which can lead to incorrect results. Both needs to be converted to number first.

`Number(yearOfEntry) >= 1988`

or convert  `Number(yearOfEntry)` at line 32 and trigger validation error straight after if it's `NaN`

Reply:
Given that `before-1988-worker-year-of-entry-to-uk` is a text input, it would be correct to use `Number.parseInt()` in this case